Sources/TuistKit/Generator/Generator.swift

```diff
@@ -167,46 +185,70 @@ class Generator: Generating {
 
     private func generateWorkspace(path: AbsolutePath) throws -> (AbsolutePath, ValueGraph) {
         // Load
-        let (graph, sideEffects) = try loadWorkspace(path: path)
+        let (graph, sideEffects) = try Signpost.measure(category: "Generator", identifier: "loadWorkspaceGraph") {
```

Comment:
Perhaps it might make more sense to make the categories some sort of a finite enum case. Too much room for typos and mistakes IMO.

Reply:
There is indeed, it can do with some improvements - enums are usually a good a choice to help with this when scoped to an individual module, here we have the `Signpost` type defined in the support library and the several clients in different modules. 

We'd need something strongly typed but also extensible such that the support library doesn't need to have awareness of all the different component categories.

Something along these lines may help reduce errors, provide enum like semantics and still allow extensibility from different modules:

```swift
// TuistSupport
public struct SignpostCategory {
    public let category: String
    public init(_ category: String) {
        self.category = category
    }
}
```

And modules can extend this type:

```swift
// TuistKit
extension SignpostCategory {
    static var generator: SignpostCategory {
        SignpostCategory("Generator")
    }
}

// Usage

try Signpost.measure(category: .generator, identifier: "loadWorkspaceGraph") {
   // ...
}
```